scripts/get_intervals.py

```diff
@@ -98,129 +69,129 @@ def get_tiling_intervals(sizes, intervalsize, shift=None):
     return intervals.loc[:, ('chrom', 'start', 'end')]
 
 
-def parse_args():
-    """Parse command line arguments.
+def get_intervals(sizesfile, intervalsize, out_dir, wg=None, val=None,
+                  holdout=None, nonpeak=None, peakfile=None, regions=None):
+    """Read chromosome sizes and generate intervals.
 
-    Return:
-        parsed argument object.
+     Args:
+         sizesfile: BED file containing sizes of each chromosome.
+         intervalsize: Size of the intervals at each row.
+         out_dir: Directory to save the output files to.
+         wg: Create intervals for the whole genome.
+         val: Chromosome to reserve for validation.
+         holdout: Chromosome to reserve for evaluation.
+         nonpeak: Ratio of nonpeak to peak intervals desired in training
+         dataset.
+         peakfile: File with clean peaks to know which intervals have non-zero
+         values. Only useful if nonpeak is greater than one.
 
-    """
-    parser = argparse.ArgumentParser(description='AtacWorks interval script')
-    parser.add_argument('--sizes', type=str,
-                        help='Path to chromosome sizes file',
-                        required=True)
-    parser.add_argument('--intervalsize', type=int, help='Interval size',
-                        required=True)
-    parser.add_argument('--out_dir', type=str, help='Directory to save \
-                        output file', required=True)
-    parser.add_argument('--prefix', type=str, help='Optional prefix to '
-                                                   'append to \
-                        output file names')
-    parser.add_argument('--shift', type=int, help='Shift between training \
-                        intervals. If not given, intervals are \
-                        non-overlapping')
-    parser.add_argument('--wg', action='store_true',
-                        help='Produce one set of intervals for whole genome')
-    parser.add_argument('--val', type=str, help='Chromosome for validation')
-    parser.add_argument('--holdout', type=str, help='Chromosome to hold out')
-    parser.add_argument('--nonpeak', type=int,
-                        help='Ratio between number of non-peak\
-                        intervals and peak intervals', default=1)
-    parser.add_argument('--peakfile', type=str,
-                        help='Path to bigWig file containing peaks. \
-                        Use when setting --nonpeak. Use peak2bw.py \
-                        to create this bigWig file.')
-    args = parser.parse_args()
-    return args
-
-
-def main():
-    """Read chromosome sizes and generate intervals."""
-    args = parse_args()
+    Returns:
+         Paths of files saved.
 
+    """
     # Read chromosome sizes
-    sizes = read_sizes(args.sizes)
+    sizes = read_sizes(sizesfile)
 
+    # Create the output dir
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
     # Generate intervals
-    if args.wg:
+    if wg:
```

Comment:
Do we need the --wg option? Could we just assume that if chromosomes are unspecified then we should use the whole genome?